[16.0][MIG] account_invoice_ubl

jbaudoux opened the pull request:
Depends on:
* https://github.com/OCA/edi/pull/893
* https://github.com/OCA/edi/pull/898

alexis-via committed:
Add modules account_invoice_ubl + base_ubl_payment

alexis-via committed:
Add partner identification hook in UBL XML party block generation

Print numbers with the approriate number of digits in the XML

alexis-via committed:
Add support for UBL 2.0 (required for e-fff)

alexis-via committed:
Update README

alexis-via committed:
Handle lang in UBL XML file generation

Commit:
8.0 Add support for partner bank matching on invoice update (#6)

Add support for partner bank matching on invoice update (before, it was only supported on invoice creation)

alexis-via committed:
Prepare v10 branch

Rename __openerp__.py to __manifest__.py and set installable to False

alexis-via committed:
Port base_business_document_import* to v10

Also port all the modules that generate the XML documents: account_invoice_ubl, account_invoice_zugferd, purchase_order_ubl and sale_order_ubl

alexis-via committed:
Rename account_invoice_zugferd to account_invoice_factur-x

Rename account_invoice_import_zugferd to account_invoice_import_factur-x
Add module to support py3o reporting engine:

alexis-via committed:
FIX crash on self._model

Add active buttons on invoice import config

alexis-via committed:
Continue port of modules for v10.0, in particular sale_order_import_* module

Fix spelling mistake and other remarks on README by Tarteo

alexis-via committed:
Port purchase_order_import* to v10.0

Add ubl invoice generation option in accounting config page

alexis-via committed:
Stop using precision_get('Account') (this decimal precision doesn't exist in v10) in account_invoice_factur-x and account_invoice_ubl

alexis-via committed:
Update test suite

alexis-via committed:
Improve + modularize generation of UBL (up-port from v8 PR)

Add script mass_invoice_import.py
Code cleanup

alexis-via committed:
Disable get_pdf() in all tests because it doesn't work in Travis

Commit:
[10.0] restore get_pdf() in tests (#31)

* sale_order_ubl + purchase_order_ubl: restore get_pdf() in tests using HttpCase

* Restore get_pdf() in tests of account_invoice_factur-x and account_invoice_ubl modules

* Update oca_dependencies.txt

alexis-via committed:
Add module account_e-invoice_generate to ensure compatibility for XML embedding in PDF between account_invoice_factur-x and account_invoice_ubl

astirpe committed:
[11.0][MIG] account_invoice_ubl

alexis-via committed:
Add support for PaymentID in UBL

Enable delivery block in UBL

alexis-via committed:
FIX is partner_shipping_id is empty

astirpe committed:
[12.0][MIG] account_invoice_ubl

astirpe committed:
[FIX] Related field not readonly

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 81.8% (9 of 11 strings)

Translation: edi-12.0/edi-12.0-account_invoice_ubl
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_ubl/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 90.9% (10 of 11 strings)

Translation: edi-12.0/edi-12.0-account_invoice_ubl
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_ubl/hr/

astirpe committed:
[13.0][MIG] account_invoice_ubl

astirpe committed:
pre-commit

astirpe committed:
Add coverage

vrenaville committed:
[FIX] test if res_ids is int, in this case change it in array

TDu committed:
[13.0][Fix] account_invoice_ubl print error

Hide the Generate UBL XML file button when the move is not posted as it
can not be generated in that state.

Also fix some crash when generating pdf for non supported invoice type
and state.

etobella committed:
[FIX] account_invoice_ubl: Missing renamed word

Commit:
account_invoice_ubl due date

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 60.0% (6 of 10 strings)

Translation: edi-13.0/edi-13.0-account_invoice_ubl
Translate-URL: https://translation.odoo-community.org/projects/edi-13-0/edi-13-0-account_invoice_ubl/nl/

Commit:
[ADD] PEPPOL UBL-DT-07: filename attribute to binary attachment

Commit:
Use invoice/sale order line taxes, not product taxes as they can be empty

Commit:
Use invoice/sale order line taxes, not product taxes as they can be empty

Commit:
[IMP] buyer reference

Commit:
BaseQuantity is actually the unit quantity, not the total quantity

fshah-initos committed:
[IMP] account_invoice_ubl: isort, black, prettier.

fshah-initos committed:
[MIG] Migrate module account_invoice_ubl to v14.0

LoisRForgeFlow committed:
[FIX] account_invoice_ubl: do not force report rendering

There is no need for this because there exists a conext in
standard (`force_report_rendering`) for that purpose and that
is correctly set in the tests of this module.

Before this change it was forced for any of pdf report
related to this module (basically invoice reports). This
could make the tests of other addons installed in the database
and not depending on this module fail.

FerranCforgeFlow committed:
[IMP] account_invoice_ubl: add hooks

* Added _ubl_get_invoice_type_code method to be inherited in case a custom invoice code is used instead of the 380/381 default ones.

hailangvn committed:
[IMP] account_invoice_ubl: fix report header, test

From nils@trobz.com:
* by default, when running in a test, _render_qweb_pdf falls back on _render_qweb_html
  (https://github.com/odoo/odoo/blob/d43b9923291dd103a80f552a38c395229a84d8db/odoo/addons/base/models/ir_actions_report.py#L736)
* but in *_ubl modules, we explicitly force_report_rendering (account_invoice_ubl/models/account_move.py)
* as such, wkhtmltopdf runs and when rendering the report content, creates 2 requests to retrieve assets (web.report_assets_common.css, web.report_assets_pdf.css)
* but when the test is running, there is a lock on the registry because the module is being installed
  (https://github.com/odoo/odoo/blob/d43b9923291dd103a80f552a38c395229a84d8db/odoo/modules/registry.py#L37)
* so the 2 assets requests are stuck in threaded mode, waiting for the lock to be released
  (https://github.com/odoo/odoo/blob/d43b9923291dd103a80f552a38c395229a84d8db/odoo/modules/registry.py#L61)
* 2 workarounds:
    * here: use HttpCase or HttpSavepointCase classes that disable the lock
      (https://github.com/odoo/odoo/blob/d43b9923291dd103a80f552a38c395229a84d8db/odoo/modules/registry.py#L684)
    * or change tests to @tagged('-at_install', 'post_install') (run after installation is complete)

oca-travis committed:
[UPD] Update account_invoice_ubl.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[IMP] update dotfiles

jbaudoux committed:
[FIX] account_invoice_ubl: Fix CreditNote

A credit note must follow a dedicated schema

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-account_invoice_ubl
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_ubl/

simahawk committed:
account_invoice_ubl: fix tests and split common

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
account_invoice_*: boost tests

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (13 of 13 strings)

Translation: edi-14.0/edi-14.0-account_invoice_ubl
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_ubl/es/

OCA-git-bot committed:
[BOT] post-merge updates

jbaudoux committed:
[IMP] account_invoice_ubl: black, isort, prettier

jbaudoux committed:
[MIG] account_invoice_ubl: Migration to 16.0

jbaudoux committed:
[FIX] account_invoice_ubl: amount & taxes

* Fix unit price, declare discount (must be declared to prevent rounding issue)

* Support non-VAT taxes: Declare non-VAT taxes as global extra charge
Peppol does not allow to have multiple taxes declaration exempt of VAT.
Regroup them in one declaration with the sum of the amounts.

jbaudoux committed:
account_invoice_ubl: pylint

jbaudoux committed:
[FIX] account_invoice_ubl: Fix OrderReference

jbaudoux committed:
[FIX] account_invoice_ubl: Fix CreditNote

A credit note must not have a DueDate

jbaudoux committed:
[FIX] account_invoice_ubl: Fix CreditNote

Do not provide payment means & terms on a credit note

lmignon committed:
[FIX] account_invoice_ubl: Fix report rendintion

The report_ref can be one of:
    - ir.actions.report id
    - ir.actions.report record
    - ir.model.data reference to ir.actions.report
    - ir.actions.report report_name
Ensures to retrieve tne report name from a report_ref accordingly.

jbaudoux commented:
@OCA/edi-maintainers Could you reopen this PR?

lmignon commented:
ping @OCA/edi-maintainers

lmignon reviewed (approved):
LGTM (Code + functionnal)

lmignon commented:
@jbaudoux a little rebase seems to be required...

jbaudoux committed:
account_invoice_ubl: exclude note lines in UBL

jbaudoux commented:
* Rebased
* Fixed non product invoice lines to exclude them from UBL. Generation was crashing with `AssertionError: name is a required arg`
* dropped merged PR from test-requirements.txt

jbaudoux committed:
account_invoice_ubl: fix tax

In case of tax exempt invoice where lines have a tax without repartition lines, declare tax based on the tax of the line

sbidoul commented:
@jbaudoux do you have information about the pro and cons of this PR vs `account_edi_xml_ubl_bis3.py` in Odoo CE for generating Peppol compliant invoices?

jbaudoux commented:
> @jbaudoux do you have information about the pro and cons of this PR vs `account_edi_xml_ubl_bis3.py` in Odoo CE for generating Peppol compliant invoices?

@sbidoul discount and taxes were not right in odoo std, at least in v16.

simahawk commented:
/ocabot merge nobump
